x/rewards/keeper/iprpc_test.go

```diff
@@ -0,0 +1,642 @@
+
+	// query with arg = mockspec
+	mockspecExpectedResults := []rewardstypes.IprpcReward{
+		{
+			Id: 2, SpecFunds: []rewardstypes.Specfund{
+				{Spec: ts.specs[0].Index, Fund: sdk.NewCoins(sdk.NewCoin(ts.BondDenom(), sdk.NewInt(100000)))},
+			},
+		},
+		{
+			Id: 3, SpecFunds: []rewardstypes.Specfund{
+				{Spec: ts.specs[0].Index, Fund: sdk.NewCoins(sdk.NewCoin(ts.BondDenom(), sdk.NewInt(100000)))},
+			},
+		},
+		{
+			Id: 4, SpecFunds: []rewardstypes.Specfund{
+				{Spec: ts.specs[0].Index, Fund: sdk.NewCoins(sdk.NewCoin(ts.BondDenom(), sdk.NewInt(100000)))},
+			},
+		},
+	}
+	res, err = ts.QueryRewardsIprpcSpecReward(ts.specs[0].Index)
+	require.NoError(t, err)
+	require.ElementsMatch(t, mockspecExpectedResults, res.IprpcRewards)
+
+	// advance a month with no providers getting rewarded this month's reward should transfer to the next month
+	// 2nd month expected funds: mockspec - 100000ulava, mock2 - 3000ulava(=2000+1000) and 500uibc
+	ts.AdvanceMonths(1)
+	ts.AdvanceEpoch()
+
+	afterMonthExpectedResults := expectedResults[1:]
+	afterMonthExpectedResults[0].SpecFunds = []rewardstypes.Specfund{
+		{Spec: ts.specs[0].Index, Fund: sdk.NewCoins(sdk.NewCoin(ts.BondDenom(), sdk.NewInt(100000)))},
+		{Spec: ts.specs[1].Index, Fund: sdk.NewCoins(
+			sdk.NewCoin(ts.BondDenom(), sdk.NewInt(3000)),
+			sdk.NewCoin(ibcDenom, sdk.NewInt(500)),
+		)},
+	}
+	res, err = ts.QueryRewardsIprpcSpecReward("")
+	require.NoError(t, err)
+	require.Len(t, res.IprpcRewards, len(afterMonthExpectedResults))
+	for i := range res.IprpcRewards {
+		require.Equal(t, afterMonthExpectedResults[i].Id, res.IprpcRewards[i].Id)
+		require.ElementsMatch(t, afterMonthExpectedResults[i].SpecFunds, res.IprpcRewards[i].SpecFunds)
+	}
+
+	// make a provider provide some service to an IPRPC eligible subscription
+	c1Acc, _ := ts.GetAccount(common.CONSUMER, 0)
+	_, p1 := ts.GetAccount(common.PROVIDER, 0)
+	relay := ts.SendRelay(p1, c1Acc, []string{ts.specs[1].Index}, 100)
+	_, err = ts.Servers.PairingServer.RelayPayment(ts.GoCtx, &relay)
+	require.NoError(t, err)
+
+	// advance month to distribute monthly rewards
+	ts.AdvanceMonths(1)
+	ts.AdvanceEpoch()
+
+	// check that the latest iprpc object has been deleted
+	afterProviderServiceExpectedResults := afterMonthExpectedResults[1:]
+	res, err = ts.QueryRewardsIprpcSpecReward("")
+	require.NoError(t, err)
+	require.ElementsMatch(t, afterProviderServiceExpectedResults, res.IprpcRewards)
+}
+
+// TestIprpcRewardObjectsUpdate tests that the IPRPC reward objects' management works as expected:
+// Scenarios:
+// 0. fund iprpc pool for 2 months, current should be 0 and first iprpc reward should be with id=1 (fund is always for the next month)
+// 1. there is no service to eligible subscriptions, month passes -> current shouldn't increment and there should be no IPRPC object
+// 2. provider provides service for eligible subscription, month passes -> current should increment by 1 and a new IPRPC reward should be created with id=current
```

Comment:
let's add a scenario fund + serve and pay providers + fund more later + serve get more rewards from both funds

Reply:
done. see https://github.com/lavanet/lava/commit/d24e2b9b06ce7e85d048887903e734174a6a55b2